cognite/client/_api_client.py

```diff
@@ -628,7 +581,7 @@ def get_partition(partition: int) -> list[dict[str, Any]]:
             return retrieved_items
 
         tasks = [(f"{i + 1}/{partitions}",) for i in range(partitions)]
-        tasks_summary = execute_tasks(get_partition, tasks, max_workers=partitions, fail_fast=True)
+        tasks_summary = execute_tasks(get_partition, tasks, max_workers=self._config.max_workers, fail_fast=True)
```

Comment:
We shouldn't accept `partitions` > `max_workers` imo., won't speed up fetching and maybe cause a higher backend load (just guessing, I'm clueless about databases). What about setting `partitions = min(partitions, max_workers)`, possibly with a warning?

Reply:
Agree. fixed!

Reply:
That broke something. Couldn't immediately see what. Let's follow up in another PR instead.